Cython/Utility/ObjectHandling.c

```diff
@@ -2553,3 +2553,101 @@ static PyObject *__Pyx_PyMethod_New(PyObject *func, PyObject *self, CYTHON_UNUSE
 #else
     #define __Pyx_PyMethod_New PyMethod_New
 #endif
+
+/////////////// UnicodeConcatInplace.proto ////////////////
+
+#if CYTHON_COMPILING_IN_PYPY || PY_MAJOR_VERSION == 2
```

Comment:
This seems more of a CPython-only optimisation than a not-PyPy thing.
(And it feels worth a separate PR.)

Reply:
OK - backed it out again. I'll submit it as a separate PR.